pennylane/ops/functions/is_commuting.py

```diff
@@ -146,120 +154,69 @@ def check_commutation_two_non_simplified_crot(operation1, operation2):
     control_control = intersection(operation1.control_wires, operation2.control_wires)
     target_target = intersection(target_wires_1, target_wires_2)
 
-    if control_control and target_target:
-        return np.all(
-            np.allclose(
-                np.matmul(operation1.matrix(), operation2.matrix()),
-                np.matmul(operation2.matrix(), operation1.matrix()),
-            )
-        )
-
-    if control_control and not target_target:
+    if control_control:
+        if target_target:
+            return _check_mat_commutation(operation1, operation2)
+        # control_control and not target_target
         return True
 
-    if not control_control and target_target:
-        return np.all(
-            np.allclose(
-                np.matmul(
-                    qml.Rot(*operation1.data, wires=operation1.wires[1]).matrix(),
-                    qml.Rot(*operation2.data, wires=operation2.wires[1]).matrix(),
-                ),
-                np.matmul(
-                    qml.Rot(*operation2.data, wires=operation2.wires[1]).matrix(),
-                    qml.Rot(*operation1.data, wires=operation1.wires[1]).matrix(),
-                ),
-            )
+    if target_target:
+        return _check_mat_commutation(
+            qml.Rot(*operation1.data, wires=operation1.wires[1]),
+            qml.Rot(*operation2.data, wires=operation2.wires[1]),
         )
     return False
 
 
-def check_simplify_identity_commutation(operation1, operation2):
+def check_simplify_identity_commutation(op):
     r"""First check that a parametric operation can be simplified to the identity operator, if it is the case then
      return the commutation relation with the second operation. If simplification is not possible, it returns None.
 
     Args:
-        operation1 (pennylane.Operation): First operation.
-        operation2 (pennylane.Operation): Second operation.
+        op(pennylane.Operation): First operation.
 
     Returns:
          Bool: True if commutation, False non-commmutation and None if not possible to simplify.
     """
-    if operation1.data and operation1.name != "U2":
-        all_zeros = np.allclose(np.mod(operation1.data, 2 * np.pi), 0)
-        if all_zeros:
-            if operation2.name not in ["Barrier", "WireCut"]:
-                return True
-            return False
+    if op.data and op.name != "U2" and np.allclose(np.mod(op.data, 2 * np.pi), 0):
+        return True
     return None
 
 
 def check_commutation_two_non_simplified_rotations(operation1, operation2):
     r"""Check that the operations are two non simplified operations. If it is the case, then it checks commutation
     for two rotations that were not simplified.
 
+    Only allowed ops are `U2`, `U3`, `Rot`, `CRot`.
```

Comment:
Why is it only allowed for these gates? At the end you use the matrix representation of the gates to check for commutativity. This could be done with any gate that defined `.matrix()`, right?

Reply:
That is a mystery I did not solve.

I guess these are the gates with multiple rotation angles and therefore can't be compared by other means like the commutation map method in `_commutes`.?

Reply:
Aha I see. For all the other gates `_commutes` is used, and for the remaining gates this function is used. But this doesn't mean that this function is only allowed for these gates, it could be used for all gates, but it is slower than `_commutes`. Should we remove this line then?